### ISSUE TITLE
Reuired, optional and default flags in response schema

### ISSUE THREAD

**Issue by ilyachase** [OAI/OpenAPI-Specification#1817]
Hi guys. I have been a little bit confused lately with these flags and its usage in **response** schema. Let's say we have pretty usual json API and it have its swagger.json with definition:

```
"NewPet": {
      "type": "object",
      "required": [
        "name"
      ],
      "properties": {
        "name": {
          "type": "string"
        },
        "info": {
          "type": "string"
        },
        "tag": {
          "type": "string",
          "default": "bar"
        }
      }
    },
```

1. If `info` field have no value and it's `optional` and does not have a default value, must it be present with `null` value, or with empty string value (`""`) or it doesn't have to be present in json API at all?
2. If `tag` field have no value and it's `optional` with `default` value equals to `bar`, must it be present with this value in every response where it have no real value, or it can be not present in response at all, and it is clients business to fill it with default value?

In other words, which one of the following responses is correct?
- `{"name": "Petty"}`
- `{"name": "Petty", "info": null}`
- `{"name": "Petty", "info": "", "tag": "bar"}`

Thanks.

**Comment by darrelmiller** [comment]
The first and third are valid.  I don't think JSON Schema accepts null as a valid value for string, so I think your second one would fail.  You only need to include properties that are mentioned in the required array.  I believe I am correct in saying that `default` has no impact on validation, it is just metadata.

**Comment by ilyachase** [comment]
Is there 100% official **response** validator (according to schema) so I would test it myself?

**Comment by ilyachase** [comment]
> Since nothing in this schema involves special OpenAPI rules, any conforming JSON Schema validator for draft-04 onward will provide the correct validation result.

Can you please give a single example? Link or git repository?

**Comment by handrews** [comment]
@ilyachase http://json-schema.org/implementations.html

**Comment by ilyachase** [comment]
Thanks.
